eth/downloader/downloader.go

```diff
@@ -990,12 +996,28 @@ func (d *Downloader) findAncestor(p *peer) (uint64, error) {
 			// Make sure the peer actually gave something valid
 			headers := packet.(*headerPack).headers
 			if len(headers) == 0 {
-				glog.V(logger.Debug).Infof("%v: empty head header set", p)
+				glog.V(logger.Warn).Infof("%v: empty head header set", p)
 				return 0, errEmptyHeaderSet
 			}
+			// Make sure the peer's reply conforms to the request
+			for i := 0; i < len(headers); i++ {
```

Comment:
I'd rather not, because I'd need to set `int64(0)` too, which looks weird. And its only one cast anyway (unless there's an error, which is exceptional case).